test/integration/test-submit-custom-header.js

```diff
@@ -31,17 +44,15 @@ var server = http.createServer(function(req, res) {
 
   form
     .on('field', function(name, value) {
-      fieldsPassed--;
-      var field = FIELDS[name];
-      assert.ok(field);
-      assert.strictEqual(value, ''+field);
+      var field = FIELDS.shift();
+      assert.strictEqual(name, field.name);
+      assert.strictEqual(value, field.value+'');
     })
     .on('file', function(name, file) {
-      fieldsPassed--;
-      var field = FIELDS[name];
-      assert.ok(field);
-      assert.strictEqual(file.name, path.basename(field.path));
-      assert.strictEqual(file.type, mime.lookup(file.name));
+      var field = FIELDS.shift();
```

Comment:
Thank you.

And strictishness is not the goal. :) Tests saved our necks quite a few times, so we're trying to be careful about them. Especially in the light of this library being used not only with end-user apps, but as part of other libraries. Biggest customer for us is `request` module, it won't be fun to break things for so many people.